lib/Graph.js

```diff
@@ -219,8 +218,10 @@ Graph.prototype.literal = function (val) {
   if (typeof val == 'undefined') return LITERAL_UNDEFINED
   if (val == null) return LITERAL_NULL
 
-  // This isn't guaranteed to be unique. We should consider switching to unique keys
-  var hash = oid.hash(val)
+  // This isn't guaranteed to be unique. Specifically, a number or string literal could collide
+  // with the hash of an object literal. Consider maintaining a separate map for number and string
+  // literals where the key is the literal.
```

Comment:
This makes me a little nervous, if OID happens to return 1-10 for numerals 1-10, then the chance of collisions with an object is much more likely. I mean, hopefully OID doesn't do that... Seems safe to try, but I think the separate maps for number/string literals is a good idea.

Reply:
Yeah -- `oid` turns numbers into big numbers, so I'm not worried about a high prob of collisions. However, the separate map may not be too bad. I'll take a small look at that.